x-pack/legacy/plugins/ml/public/overview/components/anomaly_detection_panel/anomaly_detection_panel.tsx

```diff
@@ -0,0 +1,209 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, Fragment, useState, useEffect } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiEmptyPrompt,
+  EuiLoadingSpinner,
+  EuiPanel,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { toastNotifications } from 'ui/notify';
+import { AnomalyDetectionTable } from './table';
+import { ml } from '../../../services/ml_api_service';
+import { mlResultsService } from '../../../services/results_service';
+import { getGroupsFromJobs, getStatsBarData, getJobsWithTimerange } from './utils';
+
+export interface Groups {
+  [key: string]: Group;
+}
+
+export interface Group {
+  id: string;
+  jobIds: string[];
+  docs_processed: number;
+  earliest_timestamp: number;
+  latest_timestamp: number;
+  max_anomaly_score: number | null;
+}
+
+export interface Job {
+  [key: string]: any;
+}
+
+export type Jobs = Job[];
+
+interface MaxScoresByGroup {
+  [key: string]: {
+    maxScore: number;
+    index?: number;
+  };
+}
+
+const createJobLink = '#/jobs/new_job/step/index_or_search';
+
+function getDefaultAnomalyScores(groups: Group[]): MaxScoresByGroup {
+  const anomalyScores: MaxScoresByGroup = {};
+  groups.forEach(group => {
+    anomalyScores[group.id] = { maxScore: 0 };
+  });
+
+  return anomalyScores;
+}
+
+export const AnomalyDetectionPanel: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [groups, setGroups] = useState<Groups>({});
+  const [groupsCount, setGroupsCount] = useState<number>(0);
+  const [jobsList, setJobsList] = useState<Jobs>([]);
+  const [statsBarData, setStatsBarData] = useState<any>(undefined);
+  const [errorMessage, setErrorMessage] = useState<any>(undefined);
+
+  const loadJobs = async () => {
+    setIsLoading(true);
+
+    try {
+      const jobsResult: Jobs | any = await ml.jobs.jobsSummary([]);
+      const jobsSummaryList = jobsResult.map((job: any) => {
+        job.latestTimestampSortValue = job.latestTimestampMs || 0;
+        return job;
+      });
+      const jobsGroups = getGroupsFromJobs(jobsSummaryList);
+      const jobsWithTimerange = getJobsWithTimerange(jobsSummaryList);
+      const stats = getStatsBarData(jobsSummaryList);
+      setIsLoading(false);
+      setErrorMessage(undefined);
+      setStatsBarData(stats);
+      setGroupsCount(Object.values(jobsGroups).length);
+      setGroups(jobsGroups);
+      setJobsList(jobsWithTimerange);
+      loadMaxAnomalyScores(jobsGroups);
+    } catch (e) {
+      setErrorMessage(JSON.stringify(e));
+      setIsLoading(false);
+    }
+  };
+
+  const loadMaxAnomalyScores = async (groupsObject: Groups) => {
+    const groupsList: Group[] = Object.values(groupsObject);
+    const scores = getDefaultAnomalyScores(groupsList);
+
+    try {
+      const promises: any[] = [];
```

Comment:
nit: something like 
```suggestion
const promises = groupsList
  .filter(group => group.jobIds.length > 0)
  .map((group) => {
    return mlResultsService.getOverallBucketScores(
      group.jobIds,
      1,
      group.earliest_timestamp,
      group.latest_timestamp
    );
});
```

Reply:
Yep, great suggestion for preventing the anomaly score fetch error with empty jobIds 👌